`dev-org`: Review contents of projects section (#32)
open-education-hub/methodology: chapters/develop-organize/projects/reading/README.md

### Projects

> Projects are practice items that are larger in scope and require more time, effort and resources to complete.

Author: Use "extensive" instead of "more". "More" signals a comparison.

Projects are mainly used as content for [assignments](../../../use-deliver/assignments/reading/README.md).

Typically, a project consists of a statement, support files and, where possible, validators / checkers.
The statement presents the context, goals, resources, and useful references.
Support files are pieces of digital information (source code, templates, application support files) that form a starting point for the project.
Validators / checkers are applications that automate the verification and grading of solutions.

Developing a solution in advance is strongly advised.
It provides valuable insights into the completeness of the assignment statement and accompanying support files.
Additionally, it facilitates the setup of an automated checker for closed-end assignments, guaranteeing that the submission infrastructure is properly configured to meet the requirements.
The reference solution is initially stored in a private repository during the assignment's publication period.
In certain situations, it may later become accessible to the public, serving as a point of reference for other learners.

#### Storing Projects

> Projects should be separated from other types of content within the main repository.

Author: Project are stored in the `projects/` directory of a given topic.

> Each project should have its dedicated directory, which includes:

Author: ```suggestion
Each project directory includes:
```

- **Statement**

  The project statement is the core element of any project, offering a thorough description of its requirements.

>   Typically, it includes an introduction, details about the evaluation process, and additional references.

Author: ```suggestion
  Typically, it includes an overview of goals and requirements, details about the evaluation process, and additional references.
```

Reviewer: Gabi Mocanu

Projects are practice items that are larger in scope and require more time, effort and resources to complete.
The main goal of projects is to present intense challenges to learners, and, therefore, increase their skills and knowledge.
When compared to [drills](../../drills/reading/README.md), projects aim to present a comprehensive narrative that can be divided into smaller, coherent steps.
Projects are mainly used as content for [assignments](../../../use-deliver/assignments/reading/README.md).

Typically, a project consists of a statement, support files and, where possible, validators / checkers.
The statement presents the context, goals, resources, and useful references.
Support files are pieces of digital information (source code, templates, application support files) that form a starting point for the project.
Validators / checkers are applications that automate the verification and grading of solutions.

Developing a solution in advance is strongly advised.
It provides valuable insights into the completeness of the assignment statement and accompanying support files.

> Additionally, it facilitates the setup of an automated checker for closed-end assignments, guaranteeing that the submission infrastructure is properly configured to meet our requirements.

Reviewer: ```suggestion
Additionally, it facilitates the setup of an automated checker for closed-end assignments, guaranteeing that the submission infrastructure is properly configured to meet the requirements.
```

RD: Argument why this is important.
RD: Also, I would rephrase this ^ as "must" instead of "should".
Generally, the reference solution to the project is to be kept in a private repository.

#### Storing Projects

RD: Mention before hand that a project is stored in a project directory in the content repository.

The project directory comprises of:

- **Statement**

  Serving as the central component of a project, the statement provides a comprehensive description for each task.
  RD: The word "task" is too sudden. You didn't mention the assignment is a composition of tasks.
  Often, it includes an introduction, details about the evaluation process, and additional references.
  This content is stored in Markdown format with media files.

- **Support Files**

  These act as the initial foundation for the project, establishing a consistent framework for all solutions.
  The project's skeleton may range from fill-in-the-blanks outlines and templates to straightforward guidelines.
  While the provision of skeleton files isn't feasible for every case, particularly open-ended assignments, when possible, it lays the groundwork for automation.

- **Utility Files**

> ` RD: This is a new term. Should we use "support" or should we keep "utility"?

Reviewer 2: Using `utility/` provides a means to separate generic scripts (checker, linter) from the actual working space. I think this is a good practice that does not hinder anyone from simply deleting `utility/` and moving everything to `support/`.

  These files have no direct impact on solving the assignment.
  Typically, they address peripheral aspects of the project, such as grammar checks, line/word counts, directory organization etc.
  They consist of automated scripts or configuration files intended to assisting the learner or the educator in setting up and evaluating the project.

- **Automated Evaluation Program**

>   RD: This is inconsistent: aim to use the name "program(s)" instead of "script(s)".

Reviewer 2: You are right.

  These files have no direct impact on solving the assignment.
  Typically, they address peripheral aspects of the project, such as grammar checks, line/word counts, directory organization etc.
  They consist of automated programs or configuration files intended to assisting the learner or the educator in setting up and evaluating the project.

- **Automated Evaluation Program**

  I would use checker / validator.

>   Evaluation seems a bit more for me, and involves a more subjective / human / manual part.

Author: These are my comments. Update them.

`-- checker.py
```

#### Automated Validators

It's rather pleonastic, for once, and then it gives no clue of their use.

> Do they cover skeleton and utility files?

Author: These are my comments. Update them or remove them.

`-- checker.py
```

RD: `checker.py` is an over simplification.

> You would have a checker / validator directory with checker / validation applications (Python, C etc.), reference input and output and others.

Reviewer 2: It is intended to be that way. The methodology is supposed to be generic enough to be used in most cases. The idea of operating inputs and comparing them to expected outputs is common only in Computer Science.
IMO, other fields that admit checkers (Maths, Chemistry) will only compare the student's solution with a **single** set of expected values.

To establish a strong groundwork for frontend checks in a project, using an all-in-one bundle is recommended.
In our specific context, the foundation for linting relies on [`super-linter`](https://github.com/super-linter/super-linter), and spellchecking is facilitated through [`spellcheck` GitHub action](https://github.com/rojopolis/spellcheck-github-actions), both having minimal setup.

> RD: These ^ are typically repository centric, not project specific. I don't think they belong here.

Reviewer 2: I added this as a brief example of what an automated validator might look like. To put it simply, a project requires this kind of validation, but it is not mandatory for it to ensure it.
